services/sharedchannel/sync_recv.go

```diff
@@ -290,6 +293,7 @@ func (scs *Service) updateSyncUser(patch *model.UserPatch, user *model.User, cha
 	}
 
 	user.Patch(patch)
+	user = sanitizeUserForSync(user)
```

Comment:
This looks like it might be unnecessary? `user` is the existing user from the database and `patch` doesn't contain any of the fields that need to be sanitized.

Reply:
The patch doesn't contain any security related fields, but we also don't want more innocuous stuff like turning on notifications for a fake user.